packages/git-extractor/src/routes/github/index.ts

```diff
@@ -99,68 +99,82 @@ export const getRights = async (ctx: Context) => {
-  let { username, repo, branch, commitSha } = ctx.params;
-  const userToken = getUserToken(ctx);
+  try {
+    // We get branch, etc from here because there could be slashes in a branch name,
+    // we can retrieve if this is the case from this method
+    let { username, repo, branch, commitSha } = ctx.params;
+    const userToken = getUserToken(ctx);
 
-  Sentry.setContext("repo", {
-    username,
-    repo,
-    branch,
-    commitSha,
-  });
+    Sentry.setContext("repo", {
+      username,
+      repo,
+      branch,
+      commitSha,
+    });
 
-  const path = ctx.params.path && ctx.params.path.replace("+", " ");
+    const path = ctx.params.path && ctx.params.path.replace("+", " ");
 
-  let title = `${username}/${repo}`;
-  if (path) {
-    const splittedPath = path.split("/");
-    title = title + `: ${splittedPath[splittedPath.length - 1]}`;
-  }
+    let title = `${username}/${repo}`;
+    if (path) {
+      const splittedPath = path.split("/");
+      title = title + `: ${splittedPath[splittedPath.length - 1]}`;
+    }
 
-  let isPrivate = false;
+    let isPrivate = false;
 
-  if (userToken) {
-    isPrivate = await api.isRepoPrivate(username, repo, userToken);
-  }
+    if (userToken) {
+      isPrivate = await api.isRepoPrivate(username, repo, userToken);
+    }
 
-  if (!branch) {
-    branch = await api.getDefaultBranch(username, repo, userToken);
-  }
+    if (!branch) {
+      branch = await api.getDefaultBranch(username, repo, userToken);
+    }
 
-  const downloadedFiles = await downloadRepository(
-    {
-      username,
-      repo,
-      branch,
-      path,
-    },
-    commitSha,
-    isPrivate,
-    userToken
-  );
+    const downloadedFiles = await downloadRepository(
+      {
+        username,
+        repo,
+        branch,
+        path,
+      },
+      commitSha,
+      isPrivate,
+      userToken,
+    );
 
-  if (isPrivate) {
-    api.resetShaCache({ branch, username, repo, path });
-  }
+    if (isPrivate) {
+      api.resetShaCache({ branch, username, repo, path });
+    }
 
-  console.log(
-    `Creating sandbox for ${username}/${repo}, branch: ${branch}, path: ${path}`
-  );
+    console.log(
+      `Creating sandbox for ${username}/${repo}, branch: ${branch}, path: ${path}`
+    );
 
-  const sandboxParams = await createSandbox(downloadedFiles);
+    const sandboxParams = await createSandbox(downloadedFiles);
 
-  const finalTitle = sandboxParams.title || title;
+    const finalTitle = sandboxParams.title || title;
 
-  ctx.body = {
-    ...sandboxParams,
-    // If no title is set in package.json, go for this one
-    title: finalTitle,
+    ctx.body = {
+      ...sandboxParams,
+      // If no title is set in package.json, go for this one
+      title: finalTitle,
 
-    // Privacy 2 is private, privacy 0 is public
-    privacy: isPrivate ? 2 : 0,
-  };
+      // Privacy 2 is private, privacy 0 is public
+      privacy: isPrivate ? 2 : 0,
+    };
+
+  } catch (e) {
```

Comment:
The only thing new here is that everything is now wrapped in a `try/catch` and I've added the stuff in the catch.